src/mysql.c

```diff
@@ -691,6 +694,144 @@ static int mysql_read_innodb_stats (mysql_database_t *db, MYSQL *con)
+	if (row == NULL)
+	{
+		ERROR ("mysql plugin: Failed to get wsrep statistics: "
+				"`%s' did not return any rows.", query);
+		mysql_free_result (res);
+		return (-1);
+	}
+
+	field_num = mysql_num_fields (res);
+	if (field_num < 2)
+	{
+		ERROR ("mysql plugin: Failed to get wsrep statistics: "
+				"`%s' returned less than two columns.", query);
+		mysql_free_result (res);
+		return (-1);
+	}
+
+	while ((row = mysql_fetch_row (res)))
+	{
+		char *key;
+		unsigned long long val;
+		size_t key_length;
+		key = row[0];
+		val = atoll (row[1]);
+		key_length = strlen(key);
+		if (strncmp (key, "wsrep_apply_oooe", key_length ) == 0 )
+			wsrep_apply_oooe = (derive_t) val;
+		else if (strncmp (key, "wsrep_apply_oool", key_length ) == 0 )
+			wsrep_apply_oool = (derive_t) val;
+		else if (strncmp (key, "wsrep_causal_reads", key_length ) == 0 )
+			wsrep_causal_reads = (derive_t) val;
+		else if (strncmp (key, "wsrep_commit_oooe", key_length ) == 0 )
+			wsrep_commit_oooe = (derive_t) val;
+		else if (strncmp (key, "wsrep_commit_oool", key_length ) == 0 )
+			wsrep_commit_oool = (derive_t) val;
+		else if (strncmp (key, "wsrep_flow_control_recv", key_length ) == 0 )
+			wsrep_flow_control_recv = (derive_t) val;
+		else if (strncmp (key, "wsrep_flow_control_sent", key_length ) == 0 )
+			wsrep_flow_control_sent = (derive_t) val;
+		else if (strncmp (key, "wsrep_local_bf_aborts", key_length ) == 0 )
+			wsrep_local_bf_aborts = (derive_t) val;
+		else if (strncmp (key, "wsrep_local_commits", key_length ) == 0 )
+			wsrep_local_commits = (derive_t) val;
+		else if (strncmp (key, "wsrep_local_replays", key_length ) == 0 )
+			wsrep_local_replays = (derive_t) val;
+		else if (strncmp (key, "wsrep_received", key_length ) == 0 )
+			wsrep_received = (derive_t) val;
+		else if (strncmp (key, "wsrep_received_bytes", key_length ) == 0 )
+			wsrep_received_bytes = (derive_t) val;
+		else if (strncmp (key, "wsrep_replicated", key_length ) == 0 )
+			wsrep_replicated = (derive_t) val;
+		else if (strncmp (key, "wsrep_replicated_bytes", key_length ) == 0 )
+			wsrep_replicated_bytes = (derive_t) val;
+
+		else if (strncmp (key, "wsrep_apply_window", key_length ) == 0 )
+			wsrep_apply_window = (gauge_t) val;
+		else if (strncmp (key, "wsrep_cert_deps_distance", key_length ) == 0 )
+			wsrep_cert_deps_distance = (gauge_t) val;
+		else if (strncmp (key, "wsrep_local_recv_queue", key_length ) == 0 )
+			wsrep_local_recv_queue = (gauge_t) val;
+		else if (strncmp (key, "wsrep_local_recv_queue_avg", key_length ) == 0 )
+			wsrep_local_recv_queue_avg = (gauge_t) val;
+		else if (strncmp (key, "wsrep_local_send_queue", key_length ) == 0 )
+			wsrep_local_send_queue = (gauge_t) val;
+		else if (strncmp (key, "wsrep_local_send_queue_avg", key_length ) == 0 )
+			wsrep_local_send_queue_avg = (gauge_t) val;
+		else if (strncmp (key, "wsrep_commit_window", key_length ) == 0 )
+			wsrep_commit_window = (gauge_t) val;
+	}
+
+	mysql_free_result (res);
+
```

Comment:
I might be mistaken, but wouldn't the code below submit empty values in the case one or the other of the items matched above aren't found ?

Wouldn't it be an idea to move these into `*_submit()` calls into the if/else clauses above, and avoid these numerous `wsrepo_*` variables in the process ?

It would make this consitent with how `mysql_read()` works and behaves at least, wouldn't it ?